packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -86,107 +101,49 @@ export abstract class VirtualGatewayListener {
 /**
  * Represents the properties needed to define an HTTP Listener for a VirtualGateway
  */
-class HttpGatewayListener extends VirtualGatewayListener {
-  /**
-   * Port to listen for connections on
-   *
-   * @default - 8080
-   */
-  readonly port: number;
-
-  /**
-   * Health checking strategy upstream nodes should use when communicating with the listener
-   *
-   * @default - no healthcheck
-   */
-  readonly healthCheck?: HealthCheck;
-
-  /**
-   * Protocol the listener implements
-   */
-  protected protocol: Protocol = Protocol.HTTP;
+class VirtualGatewayListenerImpl extends VirtualGatewayListener {
 
-  constructor(options: HttpGatewayListenerOptions = {}) {
+  constructor(private readonly protocol: Protocol,
+    private readonly healthCheck: HealthCheck | undefined,
+    private readonly port: number = 8080,
+    private readonly tlsCertificate: TlsCertificate | undefined) {
     super();
-    this.port = options.port ? options.port : 8080;
-    this.healthCheck = options.healthCheck;
   }
 
   /**
    * Called when the GatewayListener type is initialized. Can be used to enforce
    * mutual exclusivity
    */
-  public bind(_scope: cdk.Construct): VirtualGatewayListenerConfig {
+  public bind(scope: cdk.Construct): VirtualGatewayListenerConfig {
+    const tlsConfig = this.tlsCertificate?.bind(scope);
     return {
       listener: {
         portMapping: {
           port: this.port,
           protocol: this.protocol,
         },
         healthCheck: this.healthCheck ? renderHealthCheck(this.healthCheck, this.protocol, this.port): undefined,
+        tls: renderTls(tlsConfig),
       },
     };
   }
-}
 
-/**
-* Represents the properties needed to define an HTTP2 Listener for a VirtualGateway
-*/
-class Http2GatewayListener extends HttpGatewayListener {
-  constructor(options: HttpGatewayListenerOptions = {}) {
-    super(options);
-    this.protocol = Protocol.HTTP2;
-  }
 }
 
 /**
- * Represents the properties needed to define a GRPC Listener for Virtual Gateway
+ * Renders the TLS config for a listener
  */
-class GrpcGatewayListener extends VirtualGatewayListener {
-  /**
-   * Port to listen for connections on
-   *
-   * @default - 8080
-   */
-  readonly port: number;
-
-  /**
-   * Health checking strategy upstream nodes should use when communicating with the listener
-   *
-   * @default - no healthcheck
-   */
-  readonly healthCheck?: HealthCheck;
-
-  /**
-   * Protocol the listener implements
-   */
-  protected protocol: Protocol = Protocol.GRPC;
-
-  constructor(options: HttpGatewayListenerOptions = {}) {
-    super();
-    this.port = options.port ? options.port : 8080;
-    this.healthCheck = options.healthCheck;
-  }
+function renderTls(tlsCertificateConfig: TlsCertificateConfig | undefined): CfnVirtualGateway.VirtualGatewayListenerTlsProperty | undefined {
```

Comment:
Why accept an undefined? Just add a conditional before calling the function. 

`tls: tlsConfig ? renderTls(tlsConfig) : undefined,`

Reply:
I prefer the ternary operator here as well. I think I changed it in response to feedback.

Reply:
Couldn't find feedback that wanted it one way or another. I also prefer it as a ternary operator so I adopted it.